### PR TITLE
improved check for localStorage

### PR DESCRIPTION
**Pull request by WMDE-Fisch**
see https://de.wikipedia.org/w/index.php?title=Benutzer_Diskussion:Christoph_Fischer_(WMDE)/Gadgets/DeepCat.js&oldid=158651935

and https://github.com/Modernizr/Modernizr/blob/master/feature-detects/storage/localstorage.js#L41-L49

### CODE REVIEW

**Review by jakobw — APPROVED** [review]
Haven't tested it but looks good to me!

File: DeepCat.js [hunk 1]
```diff
@@ -507,9 +507,25 @@
 		return String( djb2Code( str ) );
 	}
 
+	/**
+	 * Function to test availability of localStorage
+	 * @see https://github.com/Modernizr/Modernizr/blob/master/feature-detects/storage/localstorage.js#L41-L49
+	 * @return {boolean}
+	 */
+	function canUseLocalStorage() {
+		var test = 'mw-deepcat-ls-test';
```
Comment (thiemowmde) [inline DeepCat.js]: I find the "ls" confusing. Just make it like Modernizr: `'mw-deepcat'`.

**Comment by thiemowmde** [comment]
Sad to see code like this, but if it's necessary, it's necessary.

I like to have JavaScript modules self-contained. But in this case I go with PerfektesChaos and suggest to use https://de.wikipedia.org/wiki/Wikipedia:Technik/Skin/JS/mw#.storage, if possible.

**Comment by WMDE-Fisch** [comment]
@thiemowmde your right, I totally forgot about that option. Thanks for the review! I now used the mediawiki implementation.

**Review by thiemowmde — APPROVED** [review]
Very nice! :+1:

**Pull request merged**